doc/contributing/bazel_notes.md

```diff
@@ -116,110 +117,144 @@ All device software, regardless of the device, is built with Bazel.
   ```
   or, remove the define/filtering flags and just append the `<device>` name like:
-  ```console
+  ```sh
   bazel test --test_output=streamed //sw/device/silicon_creator/lib:boot_data_functest_fpga_cw310
   ```
   and similarly for Verilator:
-  ```console
+  ```sh
   bazel test --test_output=streamed //sw/device/silicon_creator/lib:boot_data_functest_sim_verilator
   ```
 
 ### Chip-Level Tests
+
 * Query for all chip-level tests for FPGA:
-  ```console
+  ```sh
   bazel query 'filter(".*_fpga_cw310", kind(".*test rule", //sw/device/tests/...))'
   ```
   and for Verilator:
-  ```console
+  ```sh
   bazel query 'filter(".*_sim_verilator", kind(".*test rule", //sw/device/tests/...))'
   ```
 * Run all chip-level tests on FPGA:
-  ```console
+  ```sh
   bazel test --define DISABLE_VERILATOR_BUILD=true --test_tag_filters=cw310 //sw/device/tests/...
   ```
   and for Verilator:
-  ```console
+  ```sh
   bazel test --test_tag_filters=verilator //sw/device/tests/...
   ```
 * Run a single chip-level test on FPGA and see the output in real time:
-  ```console
+  ```sh
   bazel test \
     --define DISABLE_VERILATOR_BUILD=true
     --test_tag_filters=cw310 \
     --test_output=streamed \
     //sw/device/tests:uart_smoketest
   ```
   or, remove the define/filtering flags and just append the `<device>` name like:
-  ```console
+  ```sh
   bazel test --test_output=streamed //sw/device/tests:uart_smoketest_fpga_cw310
   ```
   and similarly for Verilator:
-  ```console
+  ```sh
   bazel test --test_output=streamed //sw/device/tests:uart_smoketest_sim_verilator
   ```
 
 # Linting Software
 
 There are several Bazel rules that enable running quality checks and fixers on code.
+They can be listed by running the following:
+```sh
+bazel query //quality:all
+```
 The subsections below describe how to use them.
 All of the tools described below are run in CI on every pull request, so it is best to run them before committing code.
 
 ## Linting C/C++ Code
+
 The OpenTitan supported linter for C/C++ files is `clang-format`.
 It can be run with Bazel as shown below.
 
 Run the following to check if all C/C++ code as been formatted correctly:
-```console
+```sh
 bazel test //quality:clang_format_check --test_output=streamed
 ```
 and run the following to fix it, if it is not formatted correctly.
-```console
+```sh
 bazel run //quality:clang_format_fix
 ```
 
+## Linting Rust Code
+
+The OpenTitan supported linter for Rust files is `cargo fmt`.
+It can be run with Bazel as shown below.
+
+Run the following to check if all C/C++ code as been formatted correctly:
+```sh
+bazel test //quality:rustfmt_check --test_output=streamed
+```
+and run the following to fix it, if it is not formatted correctly.
+```sh
+bazel run //quality:rustfmt_fix
+```
+
+## Linting Bazel files
+
+The OpenTitan bazel linter can be run with Bazel as shown below.
+
+Run the following to check if all BUILD files as been formatted correctly:
```

Comment:
```suggestion
## Formatting Bazel files

The OpenTitan project uses `buildifier` to automatically format Bazel files.

Run the following to check if all BUILD files as been formatted correctly:
```

Again, I think buildifier is more of a formatter than a linter.

[\[README.md\]](https://github.com/bazelbuild/buildtools/tree/master/buildifier):

> buildifier is a tool for formatting bazel BUILD and .bzl files with a standard convention.

Reply:
Fixed